docs/docs/spark-functions.md

```diff
@@ -0,0 +1,262 @@
+
+---
+
+#### Description
+Computes a deterministic bucket number for the given value, compatible with Iceberg table partitioning. This is useful when validating partition logic or computing bucket values externally (e.g., in ETL pipelines).
+
+#### Arguments
+
+- `value`: A primitive type (e.g., `INT`, `STRING`, `BIGINT`).
+- `num_buckets`: Positive integer indicating the number of buckets.
+
+#### Usage
+
+```sql
+SELECT system.bucket(column_value, num_buckets);
+```
+
+#### Parameters
+
+* `column_value`: Any supported primitive type (e.g., `INT`, `STRING`) to be hashed.
+* `num_buckets`: The total number of buckets (positive integer).
+
+#### Returns
+An `INT` between `0` and `num_buckets - 1`.
+
+#### Example
+
+```sql
+SELECT system.bucket('user_id_123', 8);
+-- Result: 3
+```
+
+### `system.years(value)`
+
+Returns the number of years since the Unix epoch (1970) for a given date or timestamp.
+
+#### Description
+
+Calculates `year(value) - 1970`. This is consistent with the Iceberg `years` partition transform, and can be used to compute partition keys manually.
+
+#### Arguments
+
+- `value`: A `DATE`, `TIMESTAMP`, or `TIMESTAMP_NTZ`. Other types will result in an error.
+
+#### Returns
+
+An `INT` representing the number of years from 1970. If the input is `NULL`, the result is also `NULL`.
+
+#### Examples
+
+```sql
+SELECT system.years(DATE '2017-12-01');
+-- Result: 47
+
+SELECT system.years(TIMESTAMP '1969-12-31 23:59:59');
+-- Result: -1
+```
+
+### `system.months(value)`
+
+Returns the number of months since the Unix epoch (1970-01-01) for a given date or timestamp.
+
+#### Description
+
+Calculates `12 * (year(value) - 1970) + month(value) - 1`. This is consistent with the Iceberg `months` partition transform and can be used to compute partition keys manually.
+
+#### Arguments
+
+- `value`: A `DATE`, `TIMESTAMP`, or `TIMESTAMP_NTZ`. Other types will result in an error.
+
+#### Returns
+
+An `INT` representing the number of months from January 1970. If the input is `NULL`, the result is also `NULL`.
+
+#### Examples
+
+```sql
+SELECT system.months(DATE '2017-12-01');
+-- Result: 575
+
+SELECT system.months(TIMESTAMP '1969-12-31 23:59:59');
+-- Result: -1
+
+SELECT system.months(NULL);
+-- Result: NULL
+```
+
+### `system.days(value)`
+
+Returns the number of days since the Unix epoch (1970-01-01) for a given date or timestamp.
+
+#### Description
+
+Calculates the number of days between `value` and 1970-01-01. This is consistent with the Iceberg `days` partition transform, and can be used to compute partition keys manually.
+
+#### Arguments
+
+- `value`: A `DATE`, `TIMESTAMP`, or `TIMESTAMP_NTZ`. Other types will result in an error.
+
+#### Returns
+
+An `INT` representing the number of days from 1970-01-01. If the input is `NULL`, the result is also `NULL`.
+
+#### Examples
+
+```sql
+SELECT system.days(DATE '2017-12-01');
+-- Result: 17596
+
+SELECT system.days(TIMESTAMP '1969-12-31 23:59:59');
```

Comment:
same as above. An example with 1970 as shown in TestSparkDaysFunction would be good here